test/tools/yulInterpreter/Interpreter.h

```diff
@@ -219,7 +232,30 @@ class ExpressionEvaluator: public ASTWalker
 	/// Returns the list of values of the expression.
 	std::vector<u256> values() const { return m_values; }
 
-private:
+protected:
+	void runExternalCall(evmasm::Instruction _instruction);
+	virtual std::shared_ptr<Interpreter> makeInterpreterCopy(std::map<YulString, u256> _variables = {}) const
+	{
+		return std::make_shared<Interpreter>(
+			m_state,
+			m_dialect,
+			m_scope,
+			m_enableExternalCalls,
+			m_disableMemoryTrace,
+			std::move(_variables)
+		);
+	}
+	virtual std::shared_ptr<Interpreter> makeInterpreterNew(InterpreterState& _state, Scope& _scope) const
+	{
+		return std::make_shared<Interpreter>(
```

Comment:
I guess not :laughing:

Reply:
I think I actually tried a unique pointer first :)